daml-script/runner/src/main/scala/com/digitalasset/daml/lf/engine/script/Runner.scala

```diff
@@ -400,21 +400,13 @@ class Runner(compiledPackages: CompiledPackages, script: Script.Action, timeMode
                   case ScriptF.Catch(act, handle) =>
                     run(SEApp(SEValue(act), Array(SEValue(SUnit)))).transformWith {
                       case Success(v) => Future.successful(SEValue(v))
-                      case Failure(SError.DamlEUnhandledException(ty, exc)) =>
-                        Converter
-                          .toFuture(env.translateValue(ty, exc))
-                          .flatMap { exc =>
-                            machine.setExpressionToEvaluate(
-                              SEApp(SEValue(handle), Array(SEValue(SAnyException(ty, exc))))
-                            )
-                            stepToValue().fold(
-                              Future.failed,
-                              Future.successful,
-                            )
-                          }
+                      case Failure(SError.DamlEUnhandledException(exc)) =>
+                        machine.setExpressionToEvaluate(SEApp(SEValue(handle), Array(SEValue(exc))))
+                        stepToValue()
+                          .fold(Future.failed, Future.successful)
```

Comment:
```suggestion
                        Converter.toFuture(stepToValue())
```

Reply:
Does not work because `toFuture` expects `Either[String, SValue]`  but  `stepToValue()` has type `Either[RuntimeException, SValue]`